### ISSUE TITLE
multi/handler stack trace upon interupption

### ISSUE THREAD

**Issue by void-in** [rapid7/metasploit-framework#4572]
The multi/handler when interrupted with CTRL+C will immediately print a stack trace. 

To reproduce,
```
msf > use exploit/multi/handler
msf exploit(handler) > exploit

[*] Started HTTPS reverse handler on https://0.0.0.0:443/
[*] Starting the payload handler...
^C[-] Exploit failed: Interrupt
[-] Call stack:
[-]   /opt/metasploit-framework/lib/rex/sync/thread_safe.rb:36:in `select'
[-]   /opt/metasploit-framework/lib/rex/sync/thread_safe.rb:36:in `select'
[-]   (eval):3:in `select'
[-]   /opt/metasploit-framework/modules/exploits/multi/handler.rb:57:in `exploit'
[-]   /opt/metasploit-framework/lib/msf/core/exploit_driver.rb:205:in `job_run_proc'
[-]   /opt/metasploit-framework/lib/msf/core/exploit_driver.rb:166:in `run'
[-]   /opt/metasploit-framework/lib/msf/base/simple/exploit.rb:136:in `exploit_simple'
[-]   /opt/metasploit-framework/lib/msf/base/simple/exploit.rb:171:in `exploit_simple'
[-]   /opt/metasploit-framework/lib/msf/ui/console/command_dispatcher/exploit.rb:111:in `cmd_exploit'
[-]   /opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:427:in `run_command'
[-]   /opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:389:in `block in run_single'
[-]   /opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `each'
[-]   /opt/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `run_single'
[-]   /opt/metasploit-framework/lib/rex/ui/text/shell.rb:200:in `run'
[-]   /opt/metasploit-framework/lib/metasploit/framework/command/console.rb:38:in `start'
[-]   /opt/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
[-]   ./msfconsole:48:in `<main>'
msf exploit(handler) >
```

**Comment by wchen-r7** [comment]
Yuuup, that's our bad. We made backtraces way to verbose with this pull request: https://github.com/rapid7/metasploit-framework/pull/4473

We noticed this problem and solutions are being discussed here: https://github.com/rapid7/metasploit-framework/issues/4552

**Comment by todb-r7** [comment]
A rescue and nice print message on Interrupt exceptions should be straight forward.

**Comment by wchen-r7** [comment]
Resolved in #4630.
